tests/rheology_scaled_profile.cc

```diff
@@ -0,0 +1,275 @@
+
+          std::vector<std::vector<double>> averages =
+                                          this->get_lateral_averaging().compute_lateral_averages(reference_viscosity_coordinates,
+                                              lateral_averaging_properties);
+
+          laterally_averaged_viscosity_profile.swap(averages[0]);
+
+          for (unsigned int i = 0; i < laterally_averaged_viscosity_profile.size(); ++i)
+            AssertThrow(numbers::is_finite(laterally_averaged_viscosity_profile[i]),
+                        ExcMessage("In computing depth averages, there is at"
+                                   " least one depth band that does not have"
+                                   " any quadrature points in it."
+                                   " Consider reducing number of depth layers"
+                                   " for averaging."));
+        }
+
+
+
+        double
+        compute_viscosity_scaling (const double depth) const
+        {
+          // Make maximal depth slightly larger to ensure depth < maximal_depth
+          const double maximal_depth = this->get_geometry_model().maximal_depth() *
+                                       (1.0+std::numeric_limits<double>::epsilon());
+          Assert(depth < maximal_depth, ExcInternalError());
+
+          unsigned int depth_index;
+          if (depth < reference_viscosity_coordinates.front())
+            {
+              depth_index = 0;
+            }
+          else if (depth > reference_viscosity_coordinates.back())
+            {
+              depth_index = reference_viscosity_coordinates.size() - 1;
+            }
+          else
+            {
+              depth_index = std::distance(reference_viscosity_coordinates.begin(),
+                                          std::lower_bound(reference_viscosity_coordinates.begin(),reference_viscosity_coordinates.end(),depth));
+              if (depth_index > 0)
+                depth_index -= 1;
+            }
+
+          double reference_viscosity;
+
+          if (false)
+            {
+              reference_viscosity = reference_viscosity_profile->compute_viscosity(depth);
+            }
+          else
+            {
+              reference_viscosity = reference_viscosity_profile->compute_viscosity(reference_viscosity_coordinates.at(depth_index));
+            }
+
+          const double average_viscosity = laterally_averaged_viscosity_profile[depth_index];
+
+          return reference_viscosity / average_viscosity;
+        }
+
+
+
+        void evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+                      MaterialModel::MaterialModelOutputs<dim> &out) const override
+        {
+          for (unsigned int i=0; i < in.n_evaluation_points(); ++i)
+            {
+              const double depth = this->get_geometry_model().depth(in.position[i]);
+
+              out.densities[i] = 3300.0;
+              out.viscosities[i] = 1.e21;
```

Comment:
Can you make a test that has lateral viscosity variations? Just to see that the averaging works correctly?

Reply:
Good point, I updated to have lateral variations. It works, although the results are slightly unintuitive (because the arithmetic average of 1e20 and 1e21 is not 5e20, but 5.5e20).